lib/tests/test_rewrite.rs

```diff
@@ -676,18 +677,7 @@ fn test_rebase_descendants_multiple_swap() {
         },
         hashset! {},
     );
-    let new_commit_c = assert_rebased(rebaser.rebase_next().unwrap(), &commit_c, &[&commit_d]);
-    let new_commit_e = assert_rebased(rebaser.rebase_next().unwrap(), &commit_e, &[&commit_b]);
-    assert!(rebaser.rebase_next().unwrap().is_none());
-    assert_eq!(rebaser.rebased().len(), 2);
-
-    assert_eq!(
-        *tx.mut_repo().view().heads(),
-        hashset! {
-            new_commit_c.id().clone(),
-            new_commit_e.id().clone()
-        }
-    );
+    let _ = rebaser.rebase_next(); // Panics because of the cycle
```

Comment:
This feels a bit too low-level for me, unless you insist.

There are some environments where panics just abort the program. I found one at Google where they still get `#[should_panic]` to work by running those tests in separate processes, but `catch_unwind` does not work.

Also, most of the official materials seem to recommend `should_panic`. E.g. https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic

, I don't feel entirely comfortable with how this part of Rust works. For one, I think I'd have to downcast the error to check the string. More worryingly, I also don't know how this works with threads. Will it automatically work with `nextest`? It's probably fine, but I feel like if I'm using `#[should_panic]`, it's on `nextest` and any other test framework to keep it working.

I'm hoping that checking the message is good enough. But, if you really prefer it, I can switch to `catch_unwind()`

Reply:
I don't have strong opinion. I just prefer scoped assertion over string-based matching.

> There are some environments where panics just abort the program.

Yeah, if you're writing program for bare metal embedded systems (running on Cortex-M0 for example), unwinding wouldn't be available.

Reply:
I wish this was a feature of `should_panic`. E.g., it could have a mode where you must call a function before panicking for the test to succeed. Ideally, such internals would be encapsulated in some sort of `should_panic_here` functionality.

Reply:
I think I'll keep it as is, for now. I know that some in-Google environments have trouble with unwinding, and I don't feel like researching the details at the moment.